chaincode/contracts/AssetManager.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.9;
+
+// import "hardhat/console.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "./Enums.sol";
+
+contract AssetManager is Ownable {
+    event DepositSuccesful(address user, uint256 amount, uint256 balance);
+    // event DepositFailed(
+    //     address depositor,
+    //     uint256 amount,
+    //     uint256 balance,
+    //     string reason
+    // );
+    event WithdrawalSuccesful(address user, uint256 amount, uint256 balance);
+    event WithdrawalFailed(
```

Comment:
I think the failure events are not needed. They can be changed to custom error though https://blog.soliditylang.org/2021/04/21/custom-errors/

Reply:
done